lib/saml/user_attributes/mhv.rb

```diff
@@ -2,36 +2,85 @@
 require 'saml/user_attributes/base_decorator'
 
 # TODO: remove these nocov comments when this is able to be tested.
-#:nocov:
+# :nocov:
 module SAML
   module UserAttributes
     class MHV < BaseDecorator
-      def mhv_icn
-        attributes['mhv_icn']&.first
-      end
+      PREMIUM_LOAS = %w(Premium).freeze
 
-      def mhv_profile
-        JSON.parse(attributes['mhv_profile']&.first)
+      def mhv_icn
+        attributes['mhv_icn']
       end
 
+      # NOTE: this is derived from mhv_profile which is a complex JSON object, see above
       def account_type
         mhv_profile['accountType']
       end
 
+      # NOTE: this is derived from mhv_profile which is a complex JSON object, see above
+      # QUESTION: Why is this premium user only listed to have Blue Button and not Rx or SM?
       def available_services
         mhv_profile['availableServices']
       end
 
-      def mhv_uuid
+      # NOTE: this is the same thing as mhv_correlation_id it should supercede any mhv
+      # mhv correlation id that is returned by MVI
+      def mhv_correlation_id
         attributes['mhv_uuid']
       end
 
+      def uuid
+        attributes['uuid']
+      end
+
+      # NOTE: This attribute is originated by id.me
+      def email
+        attributes['email']
+      end
+
+      # NOTE: This attribute is originated by id.me
+      def multifactor
+        attributes['mhv_multifactor']
```

Comment:
Guessing this should also just be `attributes['multifactor']` - i.e. multifactor is not a mhv/dslogon-related attribute, so it doesn't get prefixed.

Reply:
I haven't confirmed that for MHV but have for DSLogon, I would bet a large sum that mhv is the same though.

Reply:
you're probably right. Will change this too and we can verify when we're testing mhv.